accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/services/BasicSpringDataRepositoryDatabaseService.java

```diff
@@ -115,16 +116,17 @@ private void checkAccessionNotMergedOrDeprecated(ACCESSION accession) throws Acc
     }
 
     @Override
-    public List<AccessionWrapper<MODEL, String, ACCESSION>> findAllByAccession(List<ACCESSION> accessions) {
-        HashMap<ACCESSION, List<ACCESSION_ENTITY>> modelsByAccession = new HashMap<>();
-        repository.findByAccessionIn(accessions).iterator().forEachRemaining(
-                entity -> {
-                    modelsByAccession.putIfAbsent(entity.getAccession(), new ArrayList<>());
-                    modelsByAccession.get(entity.getAccession()).add(entity);
-                });
-
-        return modelsByAccession.values().stream().map(this::filterOldVersions).map(this::toModelWrapper)
-                .collect(Collectors.toList());
+    public AccessionWrapper<MODEL, String, ACCESSION> findLastVersionByAccession(ACCESSION accession)
+            throws AccessionDoesNotExistException, AccessionDeprecatedException {
+        final List<ACCESSION_ENTITY> acessionEntities = repository.findByAccession(accession);
+        try {
+            checkAccessionIsActive(acessionEntities,accession);
+        }catch (AccessionMergedException accessionMergedException){
```

Comment:
This won't do an HTTP redirect at the end. It will just return the data without communicating anything to the user.

Reply:
ya we are giving the redirect in rest controller .

Reply:
made the method to throw AccessionMergedException and adding a redirect in the controller advice.